src/main/java/com/prajina/academy/service/CourseServiceImpl.java

```diff
@@ -1,142 +1,88 @@
 package com.prajina.academy.service;
 
-import java.util.ArrayList;
-import java.util.Iterator;
 import java.util.List;
-import java.util.concurrent.atomic.AtomicLong;
 
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
+import com.prajina.academy.dao.CourseDAO;
 import com.prajina.academy.model.Course;
+import com.prajina.academy.transformer.CourseMapper;
 
 @Service("CourseService")
 public class CourseServiceImpl implements CourseService {
 
-	private static final AtomicLong counter = new AtomicLong();
-
-	// TODO Read from xls file
-	private static List<Course> courses;
-
-	static {
-		courses = getCourses();
-	}
+	@Autowired
+	private CourseDAO dao;
+	
+	@Autowired
+	private CourseMapper mapper;
 
 	@Override
+	@Transactional
```

Comment:
the advantage of transactional is that if anything fails in the logic, the database will Rollback the entire operation. It's safer for your data.